MarathonLabs/marathon
# feat(core): test filtering revamp

- add FQTN filter (com.example.MyTest#test1)
- add values list and reading values from file for most of the filters

AnnotationDataFilter is an odd one here and is not currently documented. Left as is without modifications

Using Marathonfile the following will filter the execution to only 2 specific tests:
```yaml
filteringConfiguration:
  allowlist:
    - type: "fully-qualified-test-name"
      values:
        - "com.example.MainActivityTest#testText"
        - "com.example.MainActivityAllureTest#testFailure"
  blocklist:
    - type: "package"
      values:
        - "com.example.somepackage"
        - "com.example.otherpackage"
```

Alternatively, if you need to reuse the same Marathonfile for running different tests:

```yaml
filteringConfiguration:
  allowlist:
    - type: "fully-qualified-test-name"
      file: "path/to/Marathonfilterfile" #Can be any file path, be careful with relative to workdir paths though
  blocklist:
    - type: "package"
      file: "path/to/BlockedPackages"
```

The file should contain FQTN entries on each line, empty lines will be ignored.

```bash
$ cat path/to/Marathonfilterfile
com.example.MainActivityTest#testText
com.example.MainActivityAllureTest#testFailure

$ cat path/to/BlockedPackages
com.example.somepackage
com.example.otherpackage
```

The same pattern will apply to all other filters: instead of specifying regex you can now supply either `values` or a `file` with values

Related to #470
